add a slice based key index

thomasjungblut opened the pull request.

sebheitzmann commented:
why a slice and not a map ?

thomasjungblut commented:
I need to do range scans :) We can have a linked map implementation as well, like LinkedHashMap in Java. Would cost more memory however.

sebheitzmann commented:
Yes, you're right. I don't need scans, so don't think about it. Sorry

thomasjungblut commented:
If you only ever do contains lookups, you might benefit by just reading the bloom filter instead.

sebheitzmann commented:
for the "Contain" function yes but to find the right place to get the value i still need the index.

sebheitzmann commented:
Any news about this ? Do you need helps ?

thomasjungblut commented:
I need some more time to implement and test it - but if you want to roll your own and share it, please do! 
Even the map based one would be lovely to add.

thomasjungblut commented:
added some tests, fixed some bugs. Good news, it's twice as fast to load ;-) 
The range scanning bandwidth is about the same, but I believe this limited by the value reads and seeks required.

I don't feel super comfortable to make this the default implementation just yet, but feel free to check this out.
The next iteration will be on disk, so there's no load time ;)

sebheitzmann commented:
I will check this. Thanks
